crates/sui-core/src/execution_driver.rs

```diff
@@ -80,12 +82,25 @@ pub async fn execution_process(
         };
         authority.metrics.execution_driver_dispatch_queue.dec();
 
-        // TODO: Ideally execution_driver should own a copy of epoch store and recreate each epoch.
-        let epoch_store = authority.load_epoch_store_one_call_per_task();
-
         let digest = *certificate.digest();
         trace!(?digest, "Pending certificate execution activated.");
 
+        // A single execution driver process runs across epochs. After epoch change,
+        // transactions from the previous epoch cannot be executed correctly so it is better
+        // to skip execution gracefully.
+        // This should not happen on validators, but possibe on fullnodes with local execution
+        // in tests.
+        let epoch_store = authority.load_epoch_store_one_call_per_task();
```

Comment:
can you comment on why it's safe to call this here? what is stopping epoch from changing anyway right after this line?

Reply:
I believe validator epoch change waits for pending execution to go to 0, otherwise end of epoch reverts will not work. Mismatched epoch should not be an issue with checkpoint execution on fullnode as well.